hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java

```diff
@@ -747,9 +771,9 @@ private void resubmit(ServerName oldServer, List<Action> toReplay,
     // It should be possible to have some heuristics to take the right decision. Short term,
     //  we go for one.
     boolean retryImmediately = throwable instanceof RetryImmediatelyException;
-    int nextAttemptNumber = retryImmediately ? numAttempt : numAttempt + 1;
+    int nextAttemptNumber = numAttempt + 1;
```

Comment:
How does this change our behavior @markrmiller

Reply:
I read the JIRA. That helped. I think you need a comment here on why the arbitrary '3'.... even it is just referring back to the issue. Thanks.